bai2/main.tf

```
@@ -0,0 +1,338 @@
+    
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "alb-demo-sg"
+  }
+}
+
+# tao network interface
+resource "aws_network_interface" "test1" {
+  subnet_id       = aws_subnet.subnet-public-1.id
+  private_ips     = ["10.0.1.50"]
+  security_groups = [aws_security_group.basion-demo-sg.id]
+
+}
+
+resource "aws_network_interface" "test2" {
+  subnet_id       = aws_subnet.subnet-public-2.id
+  private_ips     = ["10.0.4.51"]
+  security_groups = [aws_security_group.app-demo-sg.id]
+}
+
+#tao ElasticIP
+ resource "aws_eip" "one" {
+  vpc                       = true
+  network_interface         = aws_network_interface.test1.id
+  associate_with_private_ip = "10.0.1.50"
+  depends_on                = [aws_internet_gateway.gw-demo]
+}
+
+resource "aws_eip" "two" {
+  vpc                       = true
+  network_interface         = aws_network_interface.test2.id
+  associate_with_private_ip = "10.0.4.51"
+  depends_on                = [aws_internet_gateway.gw-demo]
+}
+
+#tao ec2 instance
+resource "aws_instance" "basion-demo-ec2" {
+  ami           = "ami-055d15d9cfddf7bd3"
+  instance_type = "t2.micro"
+  key_name = "bastion-key"
+
+  network_interface {
+    network_interface_id = aws_network_interface.test1.id
+    device_index         = 0
+  }
+  
+  credit_specification {
+    cpu_credits = "unlimited"
+  }
+  
+  tags = {
+      name = "basion-demo-ec2"
+  }
+}
+
+resource "aws_instance" "app-demo-ec2" {
+  ami           = "ami-055d15d9cfddf7bd3"
+  instance_type = "t2.micro"
+  key_name = "app-key-1"
+
+  network_interface {
+    network_interface_id = aws_network_interface.test2.id
+    device_index         = 0
+  }
+
+  credit_specification {
+    cpu_credits = "unlimited"
+  }
+  tags = {
+      name = "app-demo-ec2"
+  }
+  user_data = <<-EOF
+    sudo apt update
+    sudo apt install nginx
+    sudo apt install mysql-server 
+    sudo mysql_secure_installation
+    sudo apt install php-fpm php-mysql
+    sudo apt install php-cli unzip
+    curl -sS https://getcomposer.org/installer -o /tmp/composer-setup.php 
+    HASH=`curl -sS https://composer.github.io/installer.sig`
+    echo $HASH 
+    php -r "if (hash_file('SHA384', '/tmp/composer-setup.php') === '$HASH') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;" 
+    sudo php /tmp/composer-setup.php --install-dir=/usr/local/bin --filename=composer 
+    composer
+    sudo apt install php-mbstring php-xml php-bcmath 
+    composer create-project --prefer-dist laravel/laravel danhsach
+    php artisan
+    sudo mv ~/danhsach /var/www/danhsach 
+    sudo chown -R www-data.www-data /var/www/danhsach/storage
+    sudo chown -R www-data.www-data /var/www/danhsach/bootstrap/cache 
+    sudo ln -s /etc/nginx/sites-available/danhsach /etc/nginx/sites-enabled/ 
```

```suggestion
    sudo rm -rf /etc/nginx/sites-enabled/*
    aws s3 cp s3:///bucket-demo1126/nginx.txt /etc/nginx/sites-enabled/danhsach.conf
    aws s3 cp s3:///bucket-demo1126/env.txt /var/www/danhsach/.env 
    cd /var/www/danhsach 
    export DB_HOST=${aws_db_instance. demo_mysql_db.address}
    export DB_DATABASE=${aws_db_instance. demo_mysql_db.name}
    export DB_USERNAME=${aws_db_instance. demo_mysql_db.user}
    export DB_PASSWORD =${aws_db_instance. demo_mysql_db.password}
    sudo php artisan config:cache
```
Thử thay đổi như a chỉ